datafusion/common/src/scalar.rs

```diff
@@ -625,32 +625,36 @@ impl ScalarValue {
 
     /// whether this value is null or not.
     pub fn is_null(&self) -> bool {
-        matches!(
-            *self,
-            ScalarValue::Null
```

Comment:
You can see there are several types that are not covered (`ScalarValue::Binary` was what triggered an error). However, there was similar issues for `Interval`, etc

I have changed the code to use a `match` so if new variants of `ScalarValue` are added, this list will also need to be updated.